### ISSUE TITLE
Behavior mismatch of os.path.is_link between PyPy and CPython on Windows.

### ISSUE THREAD

**Issue by Erotemic** [pypy/pypy#4976]
I've come across what seems to be a difference between CPython and PyPy on Windows. This has to do with the behavior of `os.path.is_link` on windows junctions.

Let me start with the minimal example that illustrates the differences. 

```python
    import sys
    import pathlib
    import os

    user_dpath = pathlib.Path('~').expanduser()
    cache_dpath = user_dpath / 'AppData/Local/'
    tempdir = (cache_dpath / 'ubelt/tests/win32_pypy_issues' / sys.implementation.cache_tag)
    tempdir.mkdir(parents=True, exist_ok=True)

    print(f'tempdir={tempdir}')

    dpath = tempdir / 'dpath'
    djunk = tempdir / 'djunk'
    dpath.mkdir(parents=True, exist_ok=True)

    # Create the junction
    if not djunk.exists():
        command = 'mklink /J "{}" "{}"'.format(djunk, dpath)
        ret = os.system(command)
        assert ret == 0

    # Test if it looks like a junction
    is_link = os.path.islink(djunk)
    print(f'is_link={is_link}')
```

The code creates a temporary directory and creates an empty directory called "dpath" inside it. Then it uses `os.system` to invoke the windows `mklink` command to create a junction `djunk` that should point at "dpath". It then calls `os.path.islink` on `djunk`.

On CPython (3.10.10) it prints `is_link=False`, which is what I expect, as a junction is not a symlink.

However, on PyPy (3.9.19) it prints `is_link=True`. 

I tested on a Windows 10 box.

**Comment by devdanzin** [comment]
Confirmed on Windows 11 for PyPy 3.10 too. If we add:
```python
print(os.lstat(djunk).st_mode)
print(os.lstat(dpath).st_mode)
```

To the test code, we see that both `.st_mode` are 16895 (`0o40777`) on CPython, but the first one is 41398 (`0o120666`) on PyPy.

**Comment by mattip** [comment]
Should be fixed in 83d7d4817. Closing, please reopen or open a new issue if there are still discrepencies.
